### PR TITLE
Ensure all downloaded translations and metadata are committed

### PR DESCRIPTION
**Pull request by mokagio**
I run the `new_beta_release` lane today and, while all the translations were downloaded correctly, the were not committed.

This PR has just enough code to address the committing issues. I did not track the result running the updated lane, i.e. the translated files, because I thought I'd ship a new beta on purpose for that, so we can see the diff in the usual "merge beta n" context. You can still see the commits either by creating a new branch from this one and running `download_localized_strings_and_metadata`, or by looking at this demo PR.

### CODE REVIEW

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: If GitHub doesn't expand the lines on top of this diff, it might not be clear why this commit is here.

This commit is here to track the changes made by the `ios_download_strings_files_from_glotpress`. I haven't dug the history of how the action changed, but I'm guessing either that or its predecessor used to commit for us and we, rightly, removed it.

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: My editor told me about this unused variable 😄 

Notice that the `WordPressIntents` app extension which the variable referred to is part of the `MANUALLY_MAINTAINED_STRINGS_FILES` `Hash` which we pass to current iteration of the extractor action.

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: I decided to use the same "root" for this commit as the one above and to differentiate them using "`Localizable.strings`" vs. "Other `.strings`".

There might be a nifty way to do it all in one commit, but I actually think it might be useful to do it separately. Apart from being the result of two different actions, it might also be useful to compare how a strings changed from GlotPress in the first commit with how it was extracted in the second.

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: `Dir.glob('**/*.txt', base: metadata_directory)` didn't work. From what I understand and saw locally, there are two reasons.

First, even if we use `base:`, the paths returned by `glob` are still relative. From [the docs](https://ruby-doc.org/core-2.7.4/Dir.html#method-c-glob):

> The optional base keyword argument specifies the base directory for interpreting relative pathnames instead of the current working directory. As the results are not prefixed with the base directory name in this case, you will need to prepend the base directory name if you want real paths.

Second, `Dir.glob` returns the files that match the given pattern, but that excludes _deleted_ files. If a file is not there anymore, it can't match the given pattern. This would have made the commit incomplete.

We could argue whether our code should delete the existing translations when it doesn't find new ones. Even if we concluded that behavior is not desirable, we'd still need to account for it locally until we update it in the release toolkit. Otherwise, our local copies would remain dirty after downloading the release notes metadata if no new ones are available.

File: fastlane/Fastfile
Comment (mokagio) [reply fastlane/Fastfile]: In #18200, there are no changes to the extracted `.strings`. I verified this works locally by editing a prefixed string in `Localizable.strings` and running the action.

The reason for it is that those were the only files committed. You can see it worked well in the main release branch: https://github.com/wordpress-mobile/WordPress-iOS/commit/5fa500553e22026cb5c470a599074b003014238e . Also notice how that commits doesn't have any `Localizable.strings` files, which further shows the necessity for the other `git_commit` call above.

File: fastlane/Fastfile
Comment (AliSoftware) [reply fastlane/Fastfile]: This was already fixed by my "cleanup Fastfile" PR (which is build on top of my "split Fastfile" PR)… so committing this change here will likely lead to conflicts in both PR 😓

**Comment by AliSoftware** [comment]
Thanks for the PR. I'd have preferred that you merge #18140 an #18192 first though, or build this PR on top of #18192, because as it is we can't merge this PR without breaking those other two (and also at least one of these changes were already fixed by those)

I'll try to rebase the PR and transcribe the changes yours made in the main Fastfile to be applied in the corresponding `lanes/*.rb` instead during my day.

**Comment by AliSoftware** [comment]
@mokagio Hope you don't mind, I took the liberty to rebase your branch and force-push it on top of `tooling/fastfiles-cleanup` from https://github.com/wordpress-mobile/WordPress-iOS/pull/18192 because your changes were made on the pre-split `Fastfile`, and would have otherwise created a conflict. In practice, there would thus have been a big chance for those changes to be overwritten by the "Fastfile split" PR (which, in the eyes of git and GitHub, is just a "delete some lines and add some more" and not "cut some lines and paste the exact same ones elsewhere"… so it would probably have been hard to spot the overwrite 😉 )

I basically _manually_ recreated your changes in new commits (instead of doing a `git rebase`, which would have been a mess anyway given the split which happened since) to reproduce the same changes you did, keeping your new git commit messages as well.

---

Since this is all part of a telescoping train of PRs, I suggest the following action and merge order:

1. I'll let you approve and merge https://github.com/wordpress-mobile/WordPress-iOS/pull/18140 first
2. After that, you can merge https://github.com/wordpress-mobile/WordPress-iOS/pull/18192 next (which you already approved, but I couldn't really merge until 18140 was approved and landed first)
3. Lately, if you approve of my rework of your commits in this PR, merge this one.
   - Since you're the author for this PR and thus won't be able to approve your own PR, I've approved it on my end (after all, I agree with the changes that _you_ made in your PR, and merely moved them to the right new place) so that you can hit the merge button yourself when everything is ready
4. Finally, once all 3 have landed in `release/19.5`, you'll be able to do a new beta to test it all (including the missing translations commit that this PR aims to fix) whenever you want

```
                  ╔════════════════════════════════════════╗
                  ║ release/19.5-localization-commit-fixes ║
                  ╚══╦═════════════════════════════════════╝
          ┌──────┐   │
3)        │#18199│   │
          └──────┘   │
              ╔══════▽════════════════════╗
              ║ tooling/fastfiles-cleanup ║
              ╚══╦═══╦════════════════════╝
       ┌──────┐  │   │
2)     │#18192│  │   │
       └──────┘  │   │
          ╔══════▽═══╩══════════════════════╗
          ║ tooling/fastfile-split-refactor ║
          ╚══╦═══╦═══╦══════════════════════╝
   ┌──────┐  │   │   │
1) │#18140│  │   │   │
   └──────┘  ▼   ▼   ▼
      ╔═══════════════════════════╗
      ║       release/19.5        ║
      ╚═══════════════════════════╝
```

**Review by AliSoftware — APPROVED** [review]
Approving after reworking the PR myself to make it be built on top of #18192 (i.e. post-fastlane-split), so it's a bit cheating (kinda approving my own changes here…)

@mokagio I'll let you handle the telescoping-merging of the 3 PRs as I described in https://github.com/wordpress-mobile/WordPress-iOS/pull/18199#issuecomment-1077904769 if all look good to you.

**Comment by mokagio** [comment]
@AliSoftware your [suggestion](https://github.com/wordpress-mobile/WordPress-iOS/pull/18199#issuecomment-1077904769) in regards to the order in which to merge the PRs so that they eventually all land in `release/19.5` sounds good to me. I'll proceed with it now.

File: fastlane/Fastfile
Comment (AliSoftware) [reply fastlane/Fastfile]: @mokagio One thing I just thought about with using `git_commit` with `fastlane/metadata/**/*.txt` is: is there a risk that this might also commit especially the `review_information` file — which contains secret information, especially credentials that Apple reviewers should use to test the app? 🤔 

I think not, given [this file is `.gitignore`'d in the first place](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/.gitignore#L79), so I'm rather confident this is all good. But just wanted to flag this anyway as something to double-check next time you'll run `deliver` and all.
_See also p91TBi-4Le-p2#comment-4004 — which is not the same thing (the file did not get commited, but that did not prevent to being updated and changed in our local Macs during a past deliver _download_) but tangential._

**Pull request merged**
